Prog/New_Project_Geocode.sas

```diff
@@ -0,0 +1,565 @@
+    Anc2012 = "Advisory Neighborhood Commission (2012)"
+    Psa2012 = "Police Service Area (2012)"
+    Geo2010 = "Full census tract ID (2010): ssccctttttt"
+    Cluster_tr2000 = "Neighborhood cluster (tract-based, 2000)"
+    Cluster_tr2000_name = "Neighborhood cluster names (tract-based, 2000)"
+    zip = "ZIP code (5 digit)"
+    image_url = "OCTO property image URL"
+    Proj_addre = "Project address"
+    Proj_address_id = "Project MAR address ID"
+    Proj_image_url = "OCTO property image URL"
+    Proj_lat = "Project latitude"
+    Proj_lon = "Project longitude"
+    Proj_streetview_url = "Google Street View URL"
+    Proj_x = "Project longitude (MD State Plane Coord., NAD 1983 meters)"
+    Proj_y = "Project latitude (MD State Plane Coord., NAD 1983 meters)"
+    Proj_zip = "ZIP code (5 digit)"
+    Bldg_addre = "Building address"
+    Bldg_x = "Building longitude (MD State Plane Coord., NAD 1983 meters)"
+    Bldg_y = "Building latitude (MD State Plane Coord., NAD 1983 meters)"
+    Bldg_lon = "Building longitude"
+    Bldg_lat = "Building latitude"
+    Bldg_count = "Number of buildings for project";
+  
+  format Ward2012x $ward12a.;
+  
+  rename Ward2012x = Ward2012;
+  drop Ward2012;
+  
+run;
+
+**remove projects from geocode datasets that are no longer in project dataset**;
+
+proc sql;
+
+   create table project as
+   select *
+   from prescat.project_geocode
+   where nlihc_id in (select distinct nlihc_id from prescat.project)
+   ;
+
+quit;
+
+proc sql;
+
+   create table building as
+   select *
+   from prescat.building_geocode
+   where nlihc_id in (select distinct nlihc_id from prescat.project)
+   ;
+
+quit;
+
+**compare new datasets with old geocode datasets**;
+
+proc compare base=prescat.project_geocode compare=work.project listall maxprint=(40,32000);
+  id nlihc_id;
+run;
+
+proc compare base=prescat.building_geocode compare=work.building listall maxprint=(40,32000);
+  id Bldg_address_id;
+run;
+
+** merge new geocode files onto existing geocode files**;
+
+data prescat.Project_geocode;
+	set Project Project_geocode_a;
+	run;
+
+data prescat.Building_geocode;
+	set Building Building_geocode_a;
+	run;
+
+%File_info( data=&outlib..Project_geocode, freqvars=Ward2012 Bldg_count )
+%File_info( data=&outlib..Building_geocode, freqvars=Ward2012 )
+
+%Dup_check(
+  data=&outlib..Project_geocode,
+  by=nlihc_id,
+  id=Proj_Name Proj_addre 
+)
+
+proc print data=&outlib..Project_geocode;
+  where Ward2012 = '';
+  id nlihc_id;
+  title2 "&outlib..Project_geocode / Missing Ward";
+run;
+
+proc print data=&outlib..Building_geocode;
+  where Ward2012 = '';
+  id nlihc_id;
+  title2 "&outlib..Building_geocode / Missing Ward";
+run;
+  
+proc print data=&outlib..Building_geocode noobs;
+  id nlihc_id;
+  var ssl Ward2012 Bldg_address_id bldg_addre;
+  title2;
+run;
+
+** Macro Register_metadata - Start Definition **/
+
+%macro Register_metadata( revisions= );
+
+  %if &_remote_batch_submit %then %do;
+  
```

Comment:
For finishing, as you've done we want to only register with metadata when it's the final batch submit. Before that, here's where we want to copy the temporary versions of Project_geocode and Building_geocode to replace the permanent data sets on L:. I would do that with a Proc Sort, which ensures that the final data sets are sorted properly. 

> proc sort 
>    data=Project_geocode 
>    out=PresCat.Project_geocode (label="Preservation Catalog, Project-level geocoding info");
> by nlihc_id;
> run;
> 
> proc sort 
>    data=Building_geocode 
>    out=PresCat.Project_geocode (label="Preservation Catalog, Building-level geocoding info");
> by nlihc_id Bldg_address_id;
> run;

Note that I've added the data set labels here, which are needed for the metadata system. 

Then let's do the %File_info() for each file. 

> %File_info( data=PresCat.Project_geocode, freqvars=Ward2012 Bldg_count )
> %File_info( data=PresCat.Building_geocode, freqvars=Ward2012 )